syntax = "proto3";

import "map_renderer.proto";

package TC_proto;

message Distance {
    uint64 stop_from = 1;
    uint64 stop_to = 2;
    uint32 distance = 3;
}

message Coordinates {
	double lat = 1;
	double lng = 2;
}

message Bus {
    bool is_roundtrip = 1;
    bytes bus_name = 2;
    repeated uint64 route = 3;
}

message Stop {
    bytes stop_name = 1;
    Coordinates coordinates = 2;
    uint32 id = 3;
}

message TransportCatalogue {
    repeated Bus buses = 1;
    repeated Stop stops = 2;
    repeated Distance stops_distance = 3;
    proto_render.MapRenderer render_settings = 4;
    double bus_wait_time = 5;
    double bus_velocity = 6;
}